### PR TITLE
feat(web): create the signup view

### PR DESCRIPTION
**Pull request by Michael-Liendo**

### CODE REVIEW

File: crates/web/src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ pub fn App() -> impl IntoView {
             <Routes>
                 <Route path="/" view=Home/>
                 <Route path="/login" view=Login/>
+                <Route path="/login" view=Signup/>
```
Comment (EstebanBorai) [inline crates/web/src/lib.rs:21]: Perhaps C/P?

```suggestion
                <Route path="/signup" view=Signup/>
```

**Review by EstebanBorai — CHANGES_REQUESTED** [review]

File: crates/web/src/lib.rs (hunk 1 above)
Comment (Michael-Liendo) [reply crates/web/src/lib.rs:21]: yeah sorry, the speedy

File: crates/web/src/components/auth/register.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+use std::str::FromStr;
+
+use crate::components::text_field::{TextField, TextFieldType};
+use leptos::{
+    component, create_action, create_rw_signal, create_signal, view, IntoView, Show, SignalGet,
+    SignalGetUntracked, SignalSet,
+};
+
+use townhall_client::Client;
+use townhall_types::user::Email;
```
Comment (EstebanBorai) [inline crates/web/src/components/auth/register.rs:10]:
```suggestion
use std::str::FromStr;

use leptos::{
    component, create_action, create_rw_signal, create_signal, view, IntoView, Show, SignalGet,
    SignalGetUntracked, SignalSet,
};

use townhall_client::Client;
use townhall_types::user::Email;

use crate::components::text_field::{TextField, TextFieldType};
```

File: crates/web/src/components/modal.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+pub fn Modal(#[prop(into)] modal_status: RwSignal<bool>, children: ChildrenFn) -> impl IntoView {
+    let modal_ref = create_node_ref::<Div>();
+    let children = store_value(children);
+
+    on_click_outside(modal_ref, move |_event| modal_status.set(false));
+
+    view! {
+        <Suspense fallback=|| ()>
+            <Show when=move || modal_status.get() fallback=move || ()>
+                <div class="fixed right-0 top-0 z-10 flex h-screen w-screen flex-col items-center justify-center drop-shadow-lg backdrop-blur-[2px] bg-[#00000055]">
```
Comment (EstebanBorai) [inline crates/web/src/components/modal.rs:17]: I think `[#00000055]` could be translated to `bg-black/50`?

**Review by EstebanBorai — COMMENTED** [review]
Small comments. Looks good overall!

**Review by EstebanBorai — APPROVED** [review]

**Pull request merged**